File: audience/src/main/java/br/com/concretesolutions/audience/core/actor/SupportFragmentActor.java
package br.com.concretesolutions.audience.core.actor;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import br.com.concretesolutions.audience.core.Director;
import br.com.concretesolutions.audience.core.actor.Actor;

public abstract class SupportFragmentActor extends Fragment implements Actor {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Director.getActorRegistry().enroll(this, savedInstanceState);
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);

        if (Director.isInConfigurationChange()) {
            Director.getActorRegistry().take5(this, outState);
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Director.play().stop(Director.actorRef(this));
    }
}

File: audience-core/src/main/kotlin/io/github/victorolinasc/audience/core/actor/SupportFragmentActor.kt
package io.github.victorolinasc.audience.core.actor

import android.os.Bundle
import android.support.v4.app.Fragment

import io.github.victorolinasc.audience.core.Director

abstract class SupportFragmentActor : Fragment(), Actor {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Director.system.actorRegistry.enroll(this, savedInstanceState)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)

        if (Director.system.isInConfigurationChange) {
            Director.system.actorRegistry.take5(this, outState)
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        Director.system.stop(Director.actorRef(this))
    }
}
